Sources/PT.PM.CSharpParseTreeUst/Converter/ExpressionVisitor.cs

```diff
@@ -73,6 +73,39 @@ public override Ust VisitAnonymousObjectCreationExpression(AnonymousObjectCreati
             return result;
         }
 
+        public override Ust VisitTupleExpression(TupleExpressionSyntax node)
+        {
+            var result = new TupleCreateExpression()
+            {
+                TextSpan = node.GetTextSpan(),
+                Initializers = new List<Expression>()
+            };
+
+            for (int i = 0; i < node.Arguments.Count; i++)
+            {
+                var arg = node.Arguments[i];
+                var assignment = new AssignmentExpression
+                {
+                    Left = new IdToken($"Item{i + 1}", arg.GetTextSpan()),
+                    Right = (Expression)VisitAndReturnNullIfError(arg.Expression),
+                    TextSpan = arg.GetTextSpan()
+                };
+                result.Initializers.Add(assignment);
+
+                if (arg.NameColon != null)
+                {
+                    result.Initializers.Add(new AssignmentExpression
+                    {
+                        Left = (IdToken)VisitAndReturnNullIfError(arg.NameColon.Name),
+                        Right = (Expression)VisitAndReturnNullIfError(arg.Expression),
```

Comment:
Wouldn't something like `assignment.Right.clone()` be more efficient than rewalking the whole subtree?

Reply:
there can be quite complex structures and copying may not give a satisfying result, in contrast to the rewalking